### ISSUE TITLE
valuesOfRows does not interact well with reactive code

### ISSUE THREAD

**Issue by strindberg** [jOOQ/jOOQ#15504]
### Expected behavior

I have prepared a mcve here: https://github.com/strindberg/jooq-simple. Branch 

It seems to me that the two following constructs should be equivalent:

```kotlin
        val id = jooq.insertInto(PERSON)
            .columns(PERSON.FIRST_NAME, PERSON.LAST_NAME)
            .values(person.firstName, person.lastName).returningResult(PERSON.ID).awaitValue()
        person.addresses.forEach { address ->
            jooq.insertInto(ADDRESS)
                .columns(ADDRESS.LINE1, ADDRESS.LINE2, ADDRESS.PERSON_ID)
                .values(address.line1, address.line2, id)
                .awaitFirst()
       }
```

and 

```kotlin
        val id = jooq.insertInto(PERSON)
            .columns(PERSON.FIRST_NAME, PERSON.LAST_NAME)
            .values(person.firstName, person.lastName).returningResult(PERSON.ID).awaitValue()
        jooq.insertInto(ADDRESS)
            .columns(ADDRESS.LINE1, ADDRESS.LINE2, ADDRESS.PERSON_ID)
            .valuesOfRows(person.addresses.map { address -> row(address.line1, address.line2, id) })
            .awaitFirst()
```

However, as my mcve demonstrates, the second version doesn't seem to result in the address rows actually being inserted (or at least not inserted before the method returns).

### Actual behavior

The second version of the code above makes the unit test fail.

### Steps to reproduce the problem

Check out the code: https://github.com/strindberg/jooq-simple, branch

Run the code: `./gradlew build`

Notice that first test succeeds, but the second fails.

### jOOQ Version

3.18.5

### Database product and version

Postgresql 15

### Java Version

Kotlin 1.9, Java 17

### JDBC driver name and version (include name if unofficial driver)

postgresql-r2dbc

**Comment by lukaseder** [comment]
I didn't run your reproducer yet, though I'd say, this is due to using `awaitFirst`, no? What happens if you run `awaitLast`?

**Comment by strindberg** [comment]
I agree, `awaitLast` is more correct, of course, and I should have used that in my example, as that is what I had also tried. However, it doesn't make a difference: even with `awaitLast` the problem remains.

**Comment by lukaseder** [comment]
OK, I'll try reproducing it soon.

**Comment by strindberg** [comment]
We recently had another error in a different application, but with the same trigger: using `valuesOfRows`and reactive code . This time, the error manifested itself as a weird problem where postgres got confused about the column types in the table. Changing from `valuesOfRows` to iterating over the items and doing single inserts solved the problem, just as in the example I supplied.

I have simplified the example somewhat, and made sure `awaitLast()`is used. I can't really see how I could make the example any more reduced than this.

**Comment by lukaseder** [comment]
Thanks for your comment and for your patience. The issue is on my radar

**Comment by lukaseder** [comment]
The error goes away when you add addresses to the last person, or omit the unnecessary query:

```kotlin
    suspend fun insertPersonWithValues(person: PersonIn) {
        val id = jooq.insertInto(PERSON)
            .columns(PERSON.FIRST_NAME, PERSON.LAST_NAME)
            .values(person.firstName, person.lastName).returningResult(PERSON.ID).awaitValue()

        if (person.addresses.isNotEmpty())
            jooq.insertInto(ADDRESS)
                .columns(ADDRESS.LINE1, ADDRESS.LINE2, ADDRESS.PERSON_ID)
                .valuesOfRows(person.addresses.map { address -> row(address.line1, address.line2, id) })
                .awaitLast()
    }
```

You're effectively executing an `INSERT INTO address DEFAULT VALUES` statement, the default when you don't add any explicit `VALUES` clause. Since you did call the `VALUES` clause (via `valuesOfRows()`), we could make a case that perhaps your intention was for this to produce a no-op. After all, an `INSERT .. SELECT` that produces an empty select also effectively is a no-op (on the server).

In fact, this is what we're doing with JDBC! This query here:

```java
ctx.insertInto(T_AUTHOR)
   .columns(T_AUTHOR.ID, T_AUTHOR.LAST_NAME)
   .valuesOfRows()
   .execute();
```

Produces this log message:

```
Query is not executable  : insert into "public"."t_author"
default values
```

So, the query again produces the same SQL, but it's not being executed. So, this issue should fix the fact that `Query::isExecutable` doesn't seem to be checked in the R2DBC integration.

**Comment by lukaseder** [comment]
Fixed in versions:

- 3.19.0
- 3.18.7 (https://github.com/jOOQ/jOOQ/issues/15665)
- 3.17.16 (https://github.com/jOOQ/jOOQ/issues/15666)
- 3.16.22 (https://github.com/jOOQ/jOOQ/issues/15667)
